### PR TITLE
[RNMobile] Add block transforms integration tests for Media blocks

### PR DESCRIPTION
**Pull request by fluiddot**
## What?

Add integration tests to cover the block transform functionality of Media blocks.

## Why?

This PR is a follow-up of https://github.com/WordPress/gutenberg/pull/48792 to cover with tests the block transform functionality.

## How?

Integration tests have been added to the following Media blocks:
- Image block
- Video block
- Media & Text block
- Gallery block
- File block
- Audio block
- Cover block

### Unit tests pass
1. Check that the `Unit Tests / Mobile` job pass.

### Manual test
Follow the change introduced in https://github.com/WordPress/gutenberg/pull/48792 that enables more block transforms, it would be great to test the transformations manually to confirm that work as expected.

1. Open the editor in the browser and the app.
2. Open the transformation menu on both platforms and observe that match the same options.
**NOTE:** The web version usually displays more options because it supports more blocks. We should only check [the supported blocks in the native version](https://github.com/WordPress/gutenberg/blob/54d43c40da5a1e2a7e86f4dfe41280d64ef5f3ca/packages/block-library/src/index.native.js#L68-L114).
3. Perform the block transformations and compare the results between platforms (**check the collapsed list below**).

<details><summary><strong>Here you can check the potential block transformations for the blocks</strong></summary>

## Image 
- Gallery 
- Columns 
- Cover 
- File 
- Group 
- Media & Text 

## Video 
- Columns 
- Cover 
- File 
- Group 
- Media & Text 

## Media & Text 
- Image _(only available when contains an image)_ 
- Video _(only available when contains a video)_
- Columns 
- Cover 
- Group 

## Gallery 
- Image 
- Columns 
- Group 

## Cover 
- Image _(only available when the background is an image)_ 
- Video _(only available when the background is a video)_
- Columns 
- Media & Text 
- Group 

## File 
- Columns 
- Group 

## Audio 
- Columns 
- File 
- Group

</details>

### Testing Instructions for Keyboard

N/A

## Screenshots or screencast 
N/A

### CODE REVIEW

File: test/native/setup.js [hunk 1]
```diff
@@ -132,6 +132,15 @@ jest.mock( 'react-native-svg', () => {
...
 } );
 
+jest.mock(
+	'react-native-video',
+	() => {
+		const { forwardRef } = require( 'react' );
+		return forwardRef( mockComponent( 'ReactNativeVideo' ) );
+	},
+	{ virtual: true }
+);
```
Comment (fluiddot) [inline test/native/setup.js:142]: We need to mock `react-native-video` as it's the library used to render the video player in the Video block.

File: test/native/__mocks__/styleMock.js [hunk 2]
```diff
@@ -175,4 +175,7 @@ module.exports = {
 	innerAppender: {
 		marginLeft: 8,
 	},
+	mediaAreaPadding: {
+		width: 12,
+	},
```
Comment (fluiddot) [inline test/native/__mocks__/styleMock.js:180]: This style is used by the Media & Text block in:
https://github.com/WordPress/gutenberg/blob/49c518c2962738b941d028910c85098dea5507e0/packages/block-library/src/media-text/edit.native.js#L216

**Review by derekblank — COMMENTED** [review]
This is looking good to me, nice work! I was able to successfully test the manual transforms from the app.

I did want to note one thing, however -- when testing the transform options on the Media & Text block on mobile, I noted that "Video" is present as a transform option, which also matches the web behavior. However, in the dropdown list of block transforms for manual testing and in the [Media & Text transforms snapshot](https://github.com/WordPress/gutenberg/blob/344a14bfdc2fccf8870ed9ac27e1749383e0bce5/packages/block-library/src/media-text/test/__snapshots__/transforms.native.js.snap), the Video block is not present.

Perhaps the Video block is just missing as a test case transform option here for Media & Text, and the web and mobile behavior is correct? I just wanted to check in case the Video block should not be present as a transform option for the Media & Text block.

Screenshot 2023-03-08 at 1 24 46 pm

**Comment by fluiddot** [comment]
> I did want to note one thing, however -- when testing the transform options on the Media & Text block on mobile, I noted that "Video" is present as a transform option, which also matches the web behavior. However, in the dropdown list of block transforms for manual testing and in the [Media & Text transforms snapshot](https://github.com/WordPress/gutenberg/blob/344a14bfdc2fccf8870ed9ac27e1749383e0bce5/packages/block-library/src/media-text/test/__snapshots__/transforms.native.js.snap), the Video block is not present.
> 
> Perhaps the Video block is just missing as a test case transform option here for Media & Text, and the web and mobile behavior is correct? I just wanted to check in case the Video block should not be present as a transform option for the Media & Text block.
> 
> Screenshot 2023-03-08 at 1 24 46 pm

Good catch @derekblank. I overlooked the case where the Media & Text block has a video, in that case, it can be transformed into a Video block instead of an Image block. I'll update the test case and integration test, thanks!

**Comment by fluiddot** [comment]
@derekblank I've just noticed that, when separating the integration tests into PRs, I forgot to include tests for the Cover block. I'll include it in this PR, sorry for the inconvenience 🙇 .

**Comment by fluiddot** [comment]
Test cases for Media & Text and Cover blocks have been added in:
- https://github.com/WordPress/gutenberg/pull/48825/commits/772d6f0abc8ad395ab64b98208a992faea427e08
- https://github.com/WordPress/gutenberg/pull/48825/commits/72ba1f6d20769af47daad330f3d9e6073f807866

@derekblank let me know if you could take a quick look, thanks for the help 🙇 !

**Review by derekblank — APPROVED** [review]
LGTM! After the updates, I was able to test all transforms successfully. Thanks for adding these tests. 🚀

**Pull request merged**